src/main/java/com/uid2/shared/secure/azurecc/PolicyValidator.java

```diff
@@ -23,14 +30,21 @@ private void verifyPublicKey(MaaTokenPayload maaTokenPayload, String publicKey)
         var runtimePublicKey = maaTokenPayload.getRuntimeData().getPublicKey();
         if(!publicKey.equals(runtimePublicKey)){
             throw new AttestationClientException(
-                    String.format("Public key in payload is not match expected value. More info: runtime(%s), expected(%s)",
+                    String.format("Public key in payload does not match expected value. More info: runtime(%s), expected(%s)",
                             runtimePublicKey,
                             publicKey
                     ),
                     AttestationFailure.BAD_FORMAT);
         }
     }
 
+    private void verifyAttestationUrl(MaaTokenPayload maaTokenPayload) throws AttestationException {
+        String decodedRuntimeAttestationUrl = maaTokenPayload.getRuntimeData().getDecodedAttestationUrl();
+        if (maaTokenPayload != null && !UrlEquivalenceValidator.areUrlsEquivalent(decodedRuntimeAttestationUrl, this.attestationUrl)) {
```

Comment:
this null check is not necessary. If the payload is null, the line before will fail